app/views/devise/registrations/new.html.erb

```diff
@@ -14,13 +14,36 @@
     </div>
 
     <div class="label smallMarginTop"><%= t('devise.password') %></div>
-    <%= f.password_field :password, autocomplete: 'off', class: 'smallMarginBottom', placeholder: t('devise.password'), 'aria-label': t('devise.password') %>
-    <%= f.password_field :password_confirmation, autocomplete: 'off', placeholder: t('devise.registrations.placeholders.repeat_password'), 'aria-label': ('devise.registrations.placeholders.repeat_password') %>
+    <%= f.password_field :password, autocomplete: 'off', class: 'smallMarginBottom', placeholder: t('devise.password'), 'aria-label': t('devise.password'), id: 'firstpw' %>
+    <i onclick="show('firstpw')" class="fas fa-eye" id="display" style="position: relative; top: -65px; left: 190px; font-size: 30px; color: grey;"></i>
 
+    <%= f.password_field :password_confirmation, autocomplete: 'off', placeholder: t('devise.registrations.placeholders.repeat_password'), 'aria-label': ('devise.registrations.placeholders.repeat_password'), id:'pw' %>
+    <i onclick="show('pw')" class="fas fa-eye" id="display" style="position: relative; top: -45px; left: 190px; font-size: 30px; color: grey;"></i>
     <%= f.submit t('account.sign_up'), class: 'buttonGhostM smallMarginTop' %>
-  </div>
+  <!-- </div> -->
 <% end %>
+<script>
 
-<div class="center">
+  function show(a){
```

Comment:
We actually write components like these in React and use React on Rails to inject components into Rails views. These are [our best practices](https://github.com/ifmeorg/ifme/wiki/Frontend-Practices). This is [an example of injecting React components into Rails views](https://github.com/ifmeorg/ifme/blob/master/app/views/moments/show.html.erb#L57).

We recommend [developing](https://github.com/ifmeorg/ifme/wiki/Frontend-Practices#storybook) the component in [our Storybook](http://design.if-me.org/?path=/story/input--text) before using it in Rails.

Let me know if you need any help!

Reply:
I added [a section](https://github.com/ifmeorg/ifme/wiki/Frontend-Practices#how-to-use-react-in-rails) in our Front-end Practices doc on how to add React to Rails.

Reply:
Will do so, thank you!